docs/docs.md

```diff
@@ -0,0 +1,585 @@
+type Data = {
+  reporter: string
+  reporterId: string
+  teleop: (
+    | {
+        attempts: number
+        successes: number
+        statName: string
+      }
+    | {
+        attempted: boolean
+        succeeded: boolean
+        statName: string
+      })[]
+  auto: (
+    | {
+        attempts: number
+        successes: number
+        statName: string
+      }
+    | {
+        attempted: boolean
+        succeeded: boolean
+        statName: string
+      })[]
+}[]
+```
+
+# `/event/{eventKey}/match/{matchKey}/stats`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  alliance: "red" | "blue"
+  team: string
+  teleop: (
+    | {
+        attempts: {
+          max: number
+          avg: number
+        }
+        successes: {
+          max: number
+          avg: number
+        }
+        statName: string
+      }
+    | {
+        // total
+        attempts: number
+        // total
+        successes: number
+        statName: string
+      })[]
+  auto: (
+    | {
+        attempts: {
+          max: number
+          avg: number
+        }
+        successes: {
+          max: number
+          avg: number
+        }
+        statName: string
+      }
+    | {
+        // total
+        attempts: number
+        // total
+        successes: number
+        statName: string
+      })[]
+}[]
+```
+
+# `/event/{eventKey}/matches`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  redAlliance: string[]
+  blueAlliance: string[]
+  // UTC date - match predicted time
+  time: string
+  // example: qm3
+  key: string
+}[]
+```
+
+# `/event/{eventKey}/star`
+
+## `PUT`
+
+### Request
+
+```ts
+type Data = boolean
+```
+
+
+### Response
+
+```ts
+type Data = null
+```
+
+# `/event/{eventKey}/stats`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  team: string
+  teleop: (
+    | {
+        attempts: {
+          max: number
+          avg: number
+        }
+        successes: {
+          max: number
+          avg: number
+        }
+        statName: string
+      }
+    | {
+        // total
+        attempts: number
+        // total
+        successes: number
+        statName: string
+      })[]
+  auto: (
+    | {
+        attempts: {
+          max: number
+          avg: number
+        }
+        successes: {
+          max: number
+          avg: number
+        }
+        statName: string
+      }
+    | {
+        // total
+        attempts: number
+        // total
+        successes: number
+        statName: string
+      })[]
+}[]
+```
+
+# `/event/{eventKey}/team/{team}/info`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
+  // only if they have future matches
+  nextMatch?: {
+    redAlliance: string[]
+    blueAlliance: string[]
+    // UTC date - match predicted time
+    time: string
+    // example: qm3
+    key: string
+  }
+  rank: number
+  // ranking score
+  rankingScore: number
+}
+```
+
+# `/event/{eventKey}/team/{team}/stats/auto`
+
+## `GET`
+
+### Response
+
+```ts
+type Data = {
```

Comment:
Does this get stats for all events the team has been to? I don't really see the point of that data since robots can change so much between events, and that is really not the data the endpoints says it's pointing to. But if its just stats for that event, the event data sent back is redundant. Also this seems like it would be better and more efficient to grab all reports for a team that every single stat split up over every single match. I know we need this data for graphs but it seems like throwing it all into one array would make it unnecessarily difficult to process.

Reply:
![event_ team_ auto](https://user-images.githubusercontent.com/13206945/41543809-77ab9cd0-72cc-11e8-840a-9851f3e30bc1.png)

Yeah they can get detailed stats for either just that single event or for all events

Reply:
![event_ team_ auto](https://user-images.githubusercontent.com/13206945/41543857-92fc5f24-72cc-11e8-9ea9-d93a6722e111.png)